src/orders/components/OrderAddressFields/OrderAddressFields.tsx

```diff
@@ -0,0 +1,84 @@
+import { CustomerAddresses_user } from "@saleor/customers/types/CustomerAddresses";
+import { OrderErrorFragment } from "@saleor/fragments/types/OrderErrorFragment";
+import { SubmitPromise } from "@saleor/hooks/useForm";
+import { ConfirmButtonTransitionState } from "@saleor/macaw-ui";
+import { transformAddressToForm } from "@saleor/misc";
+import {
+  OrderDetails_order_billingAddress,
+  OrderDetails_order_shippingAddress,
+  OrderDetails_shop_countries
+} from "@saleor/orders/types/OrderDetails";
+import React from "react";
+
+import OrderCustomerAddressesEditDialog from "../OrderCustomerAddressesEditDialog";
+import {
+  AddressEditDialogVariant,
+  OrderCustomerAddressesEditDialogOutput
+} from "../OrderCustomerAddressesEditDialog/types";
+
+interface OrderAddressFieldsProps {
+  action: string;
+  isDraft: boolean;
+  customerAddressesLoading: boolean;
+  customer: CustomerAddresses_user;
+  countries: OrderDetails_shop_countries[];
+  onClose: () => void;
+  onConfirm: (data: OrderCustomerAddressesEditDialogOutput) => SubmitPromise;
+  confirmButtonState: ConfirmButtonTransitionState;
+  errors: OrderErrorFragment[];
+  orderShippingAddress: OrderDetails_order_shippingAddress;
+  orderBillingAddress: OrderDetails_order_billingAddress;
+}
+
+const OrderAddressFields: React.FC<OrderAddressFieldsProps> = ({
+  action,
+  isDraft,
+  customerAddressesLoading,
+  customer,
+  countries,
+  onClose,
+  onConfirm,
+  confirmButtonState,
+  errors,
+  orderShippingAddress,
+  orderBillingAddress
+}) => {
+  const addressFieldCommonProps = {
```

Comment:
A nitpick really, but it could be types as `Partial<OrderCustomerAddressesEditDialogProps>`